### PR TITLE
[14631] External locators XML parsing tests

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR adds tests related with the XML parsing of the external locators configuration.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A** Documentation builds and tests pass locally. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 1]
```diff
@@ -1625,6 +1626,438 @@ TEST_F(XMLProfileParserTests, external_locators_feature)
...
+                            <address>::1:2</address>
+                            <port>2235</port>
+                        </udpv6>
+                    </metatraffic_external_unicast_locators>
+                </builtin>
+            </rtps></participant></profiles>)",
+            xmlparser::XMLP_ret::XML_ERROR
+        },
+        {
+            "participant_complete_wrong_loc_no_addr",
```
Comment (jsantiago-eProsima) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:1984]: Similar to this test. What would happen if I set two \<address\> or \<port\> tags accidentally in my locator? Should it fail?

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: test/unittest/xmlparser/XMLProfileParserTests.cpp (hunk 1 above)
Comment (MiguelCompany) [reply test/unittest/xmlparser/XMLProfileParserTests.cpp:1984]: It should. I will add a test for it.

Comment (JLBuenoLopez-eProsima) [reply test/unittest/xmlparser/XMLProfileParserTests.cpp:1984]: Done in fe7ce528fc56b0c0f9a39be01044409711d7abf9

**Comment by MiguelCompany** [comment]
@jsantiago-eProsima This is ready for another review

File: test/unittest/xmlparser/XMLProfileParserTests.cpp [hunk 2]
```diff
@@ -2226,6 +2318,44 @@ TEST_F(XMLProfileParserTests, external_locators_feature)
             </data_writer></profiles>)",
             xmlparser::XMLP_ret::XML_OK
         },
+        // ------------------ data_writer negative cases
+        {
+            "writer_external_complete_duplicated_ingnore",
```
Comment (jsantiago-eProsima) [inline test/unittest/xmlparser/XMLProfileParserTests.cpp:2323]: Little typo: ignore

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: test/unittest/xmlparser/XMLProfileParserTests.cpp (hunk 2 above)
Comment (MiguelCompany) [reply test/unittest/xmlparser/XMLProfileParserTests.cpp]: Fixed on d2141ccd79c8ec0e7bf972693fe097b0f3bf5c42

**Review by jsantiago-eProsima — APPROVED** [review]

**Pull request merged**